modin/backends/pandas/query_compiler.py

```diff
@@ -321,6 +321,15 @@ def concat(self, axis, other, **kwargs):
     def _append_list_of_managers(self, others, axis, **kwargs):
         if not isinstance(others, list):
             others = [others]
+
+        if self._is_transposed:
+            return (
+                self.transpose()
+                ._append_list_of_managers(
+                    [o.transpose() for o in others], axis ^ 1, **kwargs
```

Comment:
Do we need to check if the others are already transposed or not?

Reply:
I should have added a comment here, I will add it.

The short answer is no, because we already handle the case for `transpose` on `others`.